prometheus/labels.go

```diff
@@ -165,6 +165,8 @@ func validateValuesInLabels(labels Labels, expectedNumberOfValues int) error {
 
 func validateLabelValues(vals []string, expectedNumberOfValues int) error {
 	if len(vals) != expectedNumberOfValues {
+		// The call below makes vals escape, copy them to avoid that.
+		vals := append([]string(nil), vals...)
```

Comment:
```sugestion
		vals := vals
```

Would this have the same effect?

Reply:
Interesting, I don't know.  Try it!

Reply:
I don't think it would work, as the slice still references the array that we've allocated on stack.

Reply:
Note that we already receive a copy of `vals` when this function is called, so this doesn't change anything.